packages/thumbprint-react/components/Image/components/image/index.jsx

```diff
@@ -0,0 +1,248 @@
+import React, { useState, forwardRef, useEffect } from 'react';
+import { Waypoint } from 'react-waypoint';
+import warning from 'warning';
+import PropTypes from 'prop-types';
+
+const LazyImage = ({ children, src: srcProp, sources: sourcesProp, onEnter, alt, ...rest }) => {
+    const [shouldLoadImage, setShouldLoadImage] = useState(false);
+
+    const src = shouldLoadImage ? srcProp : undefined;
+
+    const sources = shouldLoadImage ? sourcesProp : [];
+
+    return (
+        <>
+            <Waypoint
+                onEnter={() => {
+                    setShouldLoadImage(true);
+                    onEnter(true);
+                }}
+                // Start loading slightly before the user scrolls to the image
+                topOffset={-100}
+                bottomOffset={-100}
+            >
```

Comment:
Could we add `horizontal` prop here? Passed through with `horizontalLazyload` from `SmartImage`. That would unlock carousels. Tested locally by hardcoding and it worked as expected.

Reply:
I'm looking into removing React Waypoint in favor of `react-intersection-observer` with a polyfill. That would remove the need for the prop. It seems that `horizontal` exists because of performance concerns: https://github.com/civiccc/react-waypoint/issues/66#issuecomment-198870080

(If we were to add the prop, we'd also have to add it to `Avatar` and `ServiceCardImage` which is 🤮.)